src/array.c

```diff
@@ -82,7 +82,9 @@ STATIC_INLINE jl_array_t *_new_array(jl_value_t *atype, jl_genericmemory_t *mem,
 STATIC_INLINE jl_array_t *new_array(jl_value_t *atype, uint32_t ndims, size_t *dims)
 {
     size_t nel;
-    if (jl_array_validate_dims(&nel, ndims, dims) || *(size_t*)jl_tparam1(atype) != ndims)
+    if (jl_array_validate_dims(&nel, ndims, dims))
+        jl_exceptionf(jl_argumenterror_type, "invalid Array dimensions; too large for system bit width");
```

Comment:
```suggestion
        jl_exceptionf(jl_argumenterror_type, "invalid Array dimensions: too large for system bit width");
```
colon instead of semicolon here?  (also below)

Reply:
Maybe "system address width" instead of "system bit width"?